### PR TITLE
rx.match docs

### PR DESCRIPTION
**Pull request by ElijahAhianyo**

### CODE REVIEW

File: docs/library/layout/match.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+```python
+rx.match(
+    condition,
+    (case_1, component_1),
+    (case_2, component_2),
+    ...
+    default_component,
+)
+
+
```
Comment (Alek99) [inline docs/library/layout/match.md:33]: Extra spaces

File: docs/library/layout/match.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+            ("red", rx.text("red color selected")),
+            ("green", rx.text("green color selected")),
+            rx.text("Default color selected.")
+        ),
+        rx.select(
+            options,
+            placeholder="Select a color.",
+            on_change=MatchState.set_color,
+            color_schemes="twitter",
+        ),
```
Comment (Alek99) [inline docs/library/layout/match.md:64]: Should use the radix select, variant soft should shown the color better

File: docs/library/layout/match.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+def multi_match_demo():
+    return rx.vstack(
+        rx.match(
+            MultiMatchState.color,
+            ("red", "yellow", "blue", rx.text("primary color")),
+            ("green","orange", rx.text("secondary color")),
+            rx.text("neither primary nor secondary.")
+        ),
+        rx.select(
```
Comment (Alek99) [inline docs/library/layout/match.md:152]: Radix select

File: docs/library/layout/match.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+                (2, 4, 5, "blue"),
+                (6, 8, 12, "green"),
+                (10, 15, 20, 25, "orange"),
+                "black"
+                
+            )
+        ),
+        rx.hstack(
+            rx.button("increment", on_click=MatchMultiPropState.incr),
+            rx.button("decrement", on_click=MatchMultiPropState.decr),
```
Comment (Alek99) [inline docs/library/layout/match.md:244]: switch order of these increment on right and decrement left

File: docs/library/layout/match.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+    
+    def decr(self):
+        self.value -= 1
+
+
+def match_multi_prop_demo_():
+    return rx.vstack(
+        rx.text(
+            MatchMultiPropState.value, 
+            color= rx.match(
```
Comment (Alek99) [inline docs/library/layout/match.md:232]: A little hard to tell the color is changing maybe make it more apparent somehow, possible change a background color instead

**Review by Alek99 — CHANGES_REQUESTED** [review]

**Review by Alek99 — CHANGES_REQUESTED** [review]

File: docs/library/layout/match.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+```
+
+```md alert warning
+# Usage with Structural Pattern Matching
+
+The `rx.match` component is designed for structural pattern matching. If the value of your match condition evaluates to a boolean (True or False), it is recommended to use `rx.cond` instead. 
+
+Consider the following example, which is more suitable for `rx.cond`:
+
```
[the remark this answers is not in the document]
Comment (ElijahAhianyo) [reply docs/library/layout/match.md:291]: I think the bg color comes from the `md` block? Im not sure how to change that

File: docs/library/layout/match.md [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+    def decr(self):
+        self.value -= 1
+
+
+def match_multi_prop_demo_():
+    return rdxt.flex(
+        rdxt.heading(
+            MatchMultiPropState.value, 
+            color="white",
+            background_color= rx.match(
```
[the remark this answers is not in the document]
Comment (ElijahAhianyo) [reply docs/library/layout/match.md:259]: what does this comment refer to?

File: docs/library/layout/match.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Alek99) [reply docs/library/layout/match.md:291]: Then we need to separate it

File: docs/library/layout/match.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Alek99) [reply docs/library/layout/match.md:259]: > what does this comment refer to?

https://github.com/reflex-dev/reflex-web/pull/372#discussion_r1457971363

**Review by picklelo — APPROVED** [review]

**Pull request merged**